### PR TITLE
Refactor to meter address value before creation, whenever possible

### PR DESCRIPTION
**Pull request by SupunS**
Work towards https://github.com/dapperlabs/cadence-private-issues/issues/2

## Description

Just a refactor, to delay the creation of address whenever possible.

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/value.go [hunk 1]
```diff
@@ -16652,38 +16652,49 @@ func (*EphemeralReferenceValue) DeepRemove(_ *Interpreter) {
...
+			result[:common.AddressLength],
+			uint64(uint64Value),
+		)
 
-	binary.BigEndian.PutUint64(
-		result[:common.AddressLength],
-		uint64(uint64Value),
-	)
-	return NewAddressValue(memoryGauge, result)
+		return
```
Comment (robert-e-davidson3) [inline runtime/interpreter/value.go:16694]: unnecessary return

```suggestion
```

File: runtime/runtime.go [hunk 2]
```diff
@@ -1775,18 +1773,21 @@ func (r *interpreterRuntime) newCreateAccountFunction(
...
+		addressGetter := func() (address common.Address) {
+			var err error
+			wrapPanic(func() {
+				address, err = context.Interface.CreateAccount(payerAddress)
+			})
+			if err != nil {
+				panic(err)
+			}
+
+			return
```
Comment (robert-e-davidson3) [inline runtime/runtime.go:1785]: unnecessary return

```suggestion
```

**Review by robert-e-davidson3 — APPROVED** [review]
rad! job a couple of trivial suggestions

File: runtime/interpreter/value.go (hunk 1 above)
Comment (SupunS) [reply runtime/interpreter/value.go:16694]: This is actually needed - it returns the `result`.
its a shortened form of `return result`

File: runtime/runtime.go (hunk 2 above)
Comment (SupunS) [reply runtime/runtime.go:1785]: same as above

**Review by dsainati1 — APPROVED** [review]

File: runtime/interpreter/value.go (hunk 1 above)
Comment (robert-e-davidson3) [reply runtime/interpreter/value.go:16694]: TIL. Solidity does that too.

**Pull request merged**